src/jaxsim/api/ode_data.py

```diff
@@ -281,6 +281,24 @@ class PhysicsModelState(JaxsimDataclass):
         default_factory=lambda: jnp.zeros(3)
     )
 
+    def __hash__(self) -> int:
+
+        return hash(
+            (
+                hash(tuple(jnp.atleast_1d(self.joint_positions.flatten().tolist()))),
+                hash(tuple(jnp.atleast_1d(self.joint_velocities.flatten().tolist()))),
+                hash(tuple(self.base_position.flatten().tolist())),
+                hash(tuple(self.base_quaternion.flatten().tolist())),
```

Comment:
Probably I'm missing something, is the `.flatten()` necessary here?
```suggestion
                hash(tuple(jnp.atleast_1d(self.joint_positions.tolist()))),
                hash(tuple(jnp.atleast_1d(self.joint_velocities.tolist()))),
                hash(tuple(self.base_position.tolist())),
                hash(tuple(self.base_quaternion.tolist())),
```

Reply:
The operator `jnp.atleast1d(X).flatten()` is there mainly for making the hash function work on vectorized pytrees.